backend/services/esweService.ts

```diff
@@ -105,72 +105,102 @@ export default class EsweService {
       return createFlaggedContent({})
     }
 
-    const context = await this.systemOauthClient.getClientCredentialsTokens()
-    const learnerLatestAssessments = await this.curiousApi.getLearnerLatestAssessments(context, nomisId)
-
-    const compareByDateDesc = (a: curious.LearnerAssessment, b: curious.LearnerAssessment) =>
-      compareByDate(parseDate(a.qualification.assessmentDate), parseDate(b.qualification.assessmentDate))
-
-    const getLatestGrade = (
-      functionalSkillLevels: curious.LearnerLatestAssessment[],
-      qualificationType: AssessmentQualificationType
-    ): curious.LearnerAssessment => {
-      const emptyAssessment: curious.LearnerAssessment = {
-        qualification: {
-          qualificationType,
-        },
-      }
-      if (Array.isArray(functionalSkillLevels) && functionalSkillLevels.length > 0) {
-        const { qualifications } = functionalSkillLevels[0]
-        const learnerAssessment = qualifications
-          .filter((functionalSkillLevel) => functionalSkillLevel.qualification.qualificationType === qualificationType)
-          .sort(compareByDateDesc)[0]
-
-        return learnerAssessment || emptyAssessment
+    let skillLevels: Record<string, unknown> = null
```

Comment:
Personally, I would avoid having this variable. Probably the easiest solution is add a return statement at the end of the try block. I know it will duplicate the createFlaggedContent() call, but that is done elsewhere in this file.

Reply:
Yes, it might be more clear to return it.
The function can be broken down to decrease the overall size and complexity also